### PR TITLE
fix(zip): zip optimization for range requests

### PR DESCRIPTION
**Pull request by dariaterekhova-actionengine**

### CODE REVIEW

File: modules/zip/src/parse-zip/cd-file-header.ts [hunk 1]
```diff
@@ -114,13 +126,16 @@ export const parseZipCDFileHeader = async (
 export async function* makeZipCDHeaderIterator(
   fileProvider: FileProvider
 ): AsyncIterable<ZipCDFileHeader> {
-  const {cdStartOffset} = await parseEoCDRecord(fileProvider);
-  let cdHeader = await parseZipCDFileHeader(cdStartOffset, fileProvider);
-  while (cdHeader) {
+  const {cdStartOffset, cdByteSize} = await parseEoCDRecord(fileProvider);
+  const cd = new DataViewFile(
```
Comment (ibgreen) [inline modules/zip/src/parse-zip/cd-file-header.ts:130]: Nit: `cd` is very short... Maybe type out

File: modules/zip/src/parse-zip/end-of-central-directory.ts [hunk 2]
```diff
@@ -66,6 +68,7 @@ export const parseEoCDRecord = async (file: FileProvider): Promise<ZipEoCDRecord
   const zipEoCDOffset = await searchFromTheEnd(file, eoCDSignature);
 
   let cdRecordsNumber = BigInt(await file.getUint16(zipEoCDOffset + CD_RECORDS_NUMBER_OFFSET));
+  let cdByteSize = BigInt(await file.getUint32(zipEoCDOffset + CD_CD_BYTE_SIZE_OFFSET));
```
Comment (ibgreen) [inline modules/zip/src/parse-zip/end-of-central-directory.ts:71]: You should read the full chunk first, just like above. Waiting for each couple of bytes is going to destroy performance.

**Review by ibgreen — APPROVED** [review]
Good steps.

File: modules/zip/src/parse-zip/end-of-central-directory.ts (hunk 2 above)
Comment (dariaterekhova-actionengine) [reply modules/zip/src/parse-zip/end-of-central-directory.ts:71]: There's only one EoCD in every archive, so this code runs only once per slpk and doesn't affect performance at all in my observations. So I'll optimise this part in future PRs

**Pull request merged**